src/utils/proxyEndpoint.ts

```diff
@@ -17,7 +16,21 @@ export function getVisitorIdEndpoint(
   pathSuffix: string | undefined = undefined,
 ): string {
   const region = searchParams.get('region') || 'us'
-  const prefix = region === DEFAULT_REGION ? '' : `${region}.`
-  const suffix = pathSuffix ?? ''
+  let prefix = ''
+  switch (region) {
+    case 'eu':
+      prefix = 'eu.'
+      break
+    case 'ap':
+      prefix = 'ap.'
+      break
+    default:
+      prefix = ''
+      break
+  }
+  let suffix = pathSuffix ?? ''
+  if (suffix.length > 0 && !suffix.startsWith('/')) {
+    suffix = '/' + suffix
+  }
```

Comment:
```suggestion
  if (!suffix.startsWith('/')) {
    suffix = '/' + suffix
  }
```

This should be sufficient, and makes it a bit more readable.

Reply:
Agreed

Reply:
no wait. It will add `/` if the suffix is empty. I was trying to avoid that. Do you still think it should change? @TheUnderScorer

Reply:
True! I've updated the suggestion to still make it a bit simpler

Reply:
Do you think this is better than `length` check? I prefer `length` check over `if` check for empty strings because the intend is more explicit that way.

Reply:
@necipallef if suffix will be undefined, your check will cause an error.